### PR TITLE
#167313418 report article

### PR DESCRIPTION
**Pull request by salviosage**
#### What does this PR do?

- reporting an article

#### Description of Task to be completed?

- as a user, I should be able to report an article

- as a user, I should be able to see the report I have made and I should be able to delete My report

- as an admin I should be able to view all reports, reports to a specific article and I should be able to delete a report I want 

#### How should this be manually tested?

- using this Brach you can report a certain article by sending a POST `api/v1/reports/{articleSlug}`
-you can view all the report you've made by sending a GET request to `api/v1/reports`
-you can delete a report you have made by sending a DELETE request to `api/v1/reports/{reportid}`
-as admin you can view all report By sending a GET request to `api/v1/reports/all`
-as admin you can view reports for a specific article by sending a GET request to `api/v1/reports/{articleSlug}`
-as admin you can delete as well a report using the delete route UP 

#### What are the relevant pivotal tracker stories?
[#167313418](https://www.pivotaltracker.com/story/show/167313418)

#### Screenshots (if appropriate)
image
image
image

image
image
image
image
image
image
image

image
image

### CODE REVIEW

File: src/controllers/articles.controller.js [hunk 1]
```diff
@@ -73,24 +73,10 @@ class Articles {
...
-    let offset = (page - 1) * limit;
-    if (offset >= counter) {
-      offset = 0;
-    }
-
-    const { searchQueries } = req;
+    const { searchQueries, offset, limit } = req;
     const articles = await articleService.getAllArticles(offset, limit, searchQueries);
     if (!articles) {
       return res.status(200).json({ status: 200, message: 'There is no article.' });
```
Comment (Anguandia) [inline src/controllers/articles.controller.js:82]: Clarification on status codes that was raised by Leon applies here

**Review by Anguandia — DISMISSED** [review]

File: src/controllers/articles.controller.js (hunk 1 above)
Comment (Mnickii) [reply src/controllers/articles.controller.js:82]: Should be 404.

**Comment by Mnickii** [comment]
Please rebase

**Comment by salviosage** [comment]
> Please rebase

i have rebased

File: src/middlewares/likes.middleware.js [hunk 2]
```diff
@@ -6,10 +6,6 @@ const util = new Util();
 export default async (req, res, next) => {
   try {
     const user = await models.user.findOne({ where: { email: req.auth.email } });
-    if (!user) {
```
Comment (minega25) [inline src/middlewares/likes.middleware.js:9]: why was this part deleted though?

**Review by minega25 — CHANGES_REQUESTED** [review]
Great Job. I have included feedback on this?

File: src/middlewares/likes.middleware.js (hunk 2 above)
Comment (nkalyesubula) [inline src/middlewares/likes.middleware.js:9]: Could you please add back this code. The application will run crash if you access properties of a user that doesn't exist

File: src/controllers/articles.controller.js (hunk 1 above)
Comment (Anguandia) [reply src/controllers/articles.controller.js:82]: That's great

File: src/middlewares/query.check.js [hunk 3]
```diff
@@ -28,8 +28,8 @@ export const searchArticleQuerybuilder = (searchQueries) => {
 };
 
 export const checkQuery = (req, res, next) => {
-  const {
```
Comment (nkalyesubula) [inline src/middlewares/query.check.js:31]: Why is this code removed?

**Review by nkalyesubula — CHANGES_REQUESTED** [review]
Nice work Salvi. Have left some comments though

File: src/middlewares/query.check.js (hunk 3 above)
Comment (salviosage) [reply src/middlewares/query.check.js:31]: > Why is this code removed?

yes, I removed it because the offset is stated from `limit and page queries`  not from queries. @nkalyesubula  hope that I am clear to the point.

File: src/middlewares/likes.middleware.js (hunk 2 above)
Comment (salviosage) [reply src/middlewares/likes.middleware.js:9]: > Could you please add back this code. The application will run crash if you access properties of a user that doesn't exist

uhhh okay, on it

Comment (salviosage) [reply src/middlewares/likes.middleware.js:9]: > why was this part deleted though?
I removed it because it doesn't have to be there though its a case that to be sure that the user with the token is not yet deleted and for that concern, I think on delete user feature there should be the way to grab all token from deleted user and invalid it anyway not in this feature. but for this feature to be safe let me add codes back.

Comment (minega25) [reply src/middlewares/likes.middleware.js:9]: Oh okay.

**Review by minega25 — DISMISSED** [review]

File: src/middlewares/likes.middleware.js (hunk 2 above)
Comment (salviosage) [reply src/middlewares/likes.middleware.js:9]: > Could you please add back this code. The application will run crash if you access properties of a user that doesn't exist

fixed

**Comment by salviosage** [comment]
> Great Job. I have included feedback on this?

fixed

**Review by Mnickii — APPROVED** [review]
LGTM

**Comment by Mnickii** [comment]
Nice work @salviosage

**Review by nshutijonathan — APPROVED** [review]
LGTM

**Review by nkalyesubula — APPROVED** [review]

**Review by habinezadalvan — APPROVED** [review]
